src/components/Video/index.jsx

```diff
@@ -29,76 +49,168 @@ function Video({ video, playerRef, setCurrentVideoTime }) {
     return () => window.removeEventListener("resize", handleResize);
   }, []);
 
+  useEffect(() => {
+    function getVideoSize() {
+      const rect = elementRef.current.getBoundingClientRect();
+
+      videoLeft.current = rect.left;
+      videoTop.current = rect.top;
+
+      setVideoRight(() => Math.floor(rect.right));
+      setVideoBottom(() => Math.floor(rect.bottom));
+    }
+
+    getVideoSize();
+  }, [isCapturing]);
+
   function handleMoreClick() {
     setShowMore((prev) => !prev);
   }
 
+  function handleMouseDown(event) {
+    startX.current = event.clientX - videoLeft.current;
+    startY.current = event.clientY - videoTop.current;
+  }
+
+  async function handleMouseUp(event) {
+    setIsLoading(() => true);
+    setExtractedCode(() => null);
+
+    endX.current = event.clientX - videoLeft.current;
+    endY.current = event.clientY - videoTop.current;
+
+    const clientCoordinate = {
+      left: Math.min(startX.current, endX.current),
+      top: Math.min(startY.current, endY.current),
+    };
+
+    const videoSize = {
+      width: videoRight - videoLeft.current,
+      height: videoBottom - videoTop.current,
+    };
+
+    const captureBoxSize = {
+      width:
+        Math.max(startX.current, endX.current) -
+        Math.min(startX.current, endX.current),
+      height:
+        Math.max(startY.current, endY.current) -
+        Math.min(startY.current, endY.current),
+    };
+
+    const response = await axios
+      .post(`${import.meta.env.VITE_BASE_URL}/extraction`, {
+        clientCoordinate,
+        videoSize,
+        captureBoxSize,
+        currentVideoTime,
+        youtubeVideoId: video.youtubeVideoId,
+      })
+      .then((res) => {
+        setIsLoading(() => false);
```

Comment:
```suggestion
        setIsLoading(false);
        setIsLoading((prev) => !prev);
```
파라미터값이 따로 없어 이렇게 바꿀 수 있을 것 같아요, 아니면 아래처럼 이전 값의 not operator를 넣어도 좋을 것 같습니다.

Reply:
앗 정확하게 로딩 상태 컨트롤 하기위해 setIsLoading(false)로 바꾸겠습니다!